File: src/types/metadata.ts
export interface MetadataInput {
    file_path: string;
    title?: string;
    artist?: string;
    album?: string;
    album_artist?: string;
    composer?: string;
    genre?: string;
    year?: number;
    track_number?: number;
    total_tracks?: number;
    disc_number?: number;
    total_discs?: number;
    is_compilation?: boolean;
    cover_image_data?: string;
    comment?: string;
}

export interface MetadataOutput {
    title?: string;
    artist?: string;
    album?: string;
    album_artist?: string;
    composer?: string;
    genre?: string;
    year?: number;
    track_number?: number;
    total_tracks?: number;
    disc_number?: number;
    total_discs?: number;
    is_compilation: boolean;
    cover_image_data?: string;
    comment?: string;
}

export type MetadataStatus = 'idle' | 'reading' | 'writing' | 'success' | 'error';

File: src-tauri/src/types/metadata.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Deserialize, Serialize)]
pub struct MetadataInput {
    pub file_path: String,
    pub title: Option<String>,
    pub artist: Option<String>,
    pub album: Option<String>,
    pub album_artist: Option<String>,
    pub composer: Option<String>,
    pub genre: Option<String>,
    pub year: Option<u32>,
    pub track_number: Option<u32>,
    pub total_tracks: Option<u32>,
    pub disc_number: Option<u32>,
    pub total_discs: Option<u32>,
    pub is_compilation: Option<bool>,
    pub cover_image_data: Option<String>,
    pub comment: Option<String>,
}

#[derive(Debug, Clone, Serialize)]
pub struct MetadataOutput {
    pub title: Option<String>,
    pub artist: Option<String>,
    pub album: Option<String>,
    pub album_artist: Option<String>,
    pub composer: Option<String>,
    pub genre: Option<String>,
    pub year: Option<u32>,
    pub track_number: Option<u32>,
    pub total_tracks: Option<u32>,
    pub disc_number: Option<u32>,
    pub total_discs: Option<u32>,
    pub is_compilation: bool,
    pub cover_image_data: Option<String>,
    pub comment: Option<String>,
}
